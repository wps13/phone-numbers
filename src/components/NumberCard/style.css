.numberCard {
  border: 1px solid gray;
  border-radius: 5px;
  padding: 10px;
  margin: 10px auto;
  width: 100%;
  max-width: 450px;
  min-width: 200px;
}

.numberCard:hover {
  background-color: #9ea8b8;
  color: #fff;
  transition: 300ms ease-in;
}

.numberCard__number span,
.numberCard__cost span {
  font-weight: 600;
  color: #41454d;
}
