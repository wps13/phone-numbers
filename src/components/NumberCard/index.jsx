import React from "react";

import "./style.css";

export default ({ number, cost }) => {
  return (
    <div data-testid="number-card" className="numberCard">
      <p className="numberCard__number">
        <span>Number: </span>
        {number}
      </p>
      <p className="numberCard__cost">
        <span> cost: </span>
        {cost}
      </p>
    </div>
  );
};
