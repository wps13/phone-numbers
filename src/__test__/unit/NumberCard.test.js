import React from "react";

import ReactDOM from "react-dom";

import NumberCard from "../../components/NumberCard";

import { render, cleanup } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);

it("render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<NumberCard />, div);
});

it("render card correctly", () => {
  const { getByTestId } = render(<NumberCard number="555000001" cost="1.01" />);
  expect(getByTestId("number-card")).toHaveTextContent(
    "Number: 555000001 cost: 1.01"
  );
});

it("matches snapshot", () => {
  const tree = renderer.create(<NumberCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
