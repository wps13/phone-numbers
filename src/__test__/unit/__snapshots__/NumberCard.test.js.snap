// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="numberCard"
  data-testid="number-card"
>
  <p
    className="numberCard__number"
  >
    <span>
      Number: 
    </span>
  </p>
  <p
    className="numberCard__cost"
  >
    <span>
       cost: 
    </span>
  </p>
</div>
`;
